ophyd/v2/core.py

```diff
@@ -416,82 +418,135 @@ def close(self):
         self._listeners.remove(self)
 
 
+class SimConverter(Generic[T]):
+    def value(self, value: T) -> T:
+        return value
+
+    def write_value(self, value: T) -> T:
+        return value
+
+    def descriptor(self, source: str, value) -> Descriptor:
+        dtype = primitive_dtypes[type(value)]
+        return dict(source=source, dtype=dtype, shape=[])
+
+    def make_initial_value(self, datatype: Optional[Type[T]]) -> T:
+        if datatype is None:
+            return cast(T, None)
+
+        return datatype()
+
+
+class SimArrayConverter(SimConverter):
+    def descriptor(self, source: str, value) -> Descriptor:
+        return dict(source=source, dtype="array", shape=[len(value)])
+
+    def make_initial_value(self, datatype: Optional[Type[T]]) -> T:
+        if datatype is None:
+            return cast(T, None)
+
+        if get_origin(datatype) == abc.Sequence:
+            return cast(T, [])
+
+        return cast(T, datatype(shape=0))
+
+
+@dataclass
+class SimEnumConverter(SimConverter):
+    enum_class: Type[Enum]
+
+    def write_value(self, value: Union[Enum, str]):
+        if isinstance(value, Enum):
+            return value.value
+        else:
+            return value
```

Comment:
As we discussed, I don't think we should do any actual conversions in the converter

Reply:
I've decided I'd like to keep simple conversions; the reason being, we can have a SimSignalBackend that fails with a 'Disconnected' message if `connect()` hasn't been called, in the same way the other backends do.

The nice thing about these simulated converter classes is we can keep logic that is separate to each 'type' in its own class rather than munging them together in the SimSignalBackend.